internal/p2p/peermanager.go

```diff
@@ -815,23 +934,89 @@ func (m *PeerManager) Advertise(peerID types.NodeID, limit uint16) []NodeAddress
 		addresses = append(addresses, m.options.SelfAddress)
 	}
 
-	for _, peer := range m.store.Ranked() {
+	var numAddresses int
+	var attempts int
+
+	ranked := m.store.Ranked()
+	seenAddresses := map[NodeAddress]struct{}{}
+
+	// get the total number of possible addresses
+	for _, peer := range ranked {
 		if peer.ID == peerID {
 			continue
 		}
 
-		for nodeAddr, addressInfo := range peer.AddressInfo {
+		for addr := range peer.AddressInfo {
+			if _, ok := m.options.PrivatePeers[addr.NodeID]; !ok {
+				numAddresses++
+			}
+		}
+	}
+
```

Comment:
I believe we want `TotalScore` instead of `NumAddresses`.

Reply:
I don't get why total score is useful here? this is just so we know how many addresses we have and can abort the later loop in the "small number of peers case"

Reply:
oh, now I get it.